freeipa/src/main/java/com/sequenceiq/freeipa/converter/cloud/StackToCloudStackConverter.java

```diff
@@ -320,6 +323,20 @@ private Security buildSecurity(InstanceGroup ig) {
         return new Security(rules, ig.getSecurityGroup().getSecurityGroupIds(), true);
     }
 
+    public CloudStack updateWithVerticalScaleRequest(CloudStack cloudStack, FreeIpaVerticalScaleRequest request) {
+        cloudStack.getGroups().forEach(group -> {
+            InstanceTemplateRequest template = request.getTemplate();
+            if (group.getName().equals(request.getGroup())) {
+                group.getInstances()
+                        .stream()
+                        .filter(instance -> template != null)
+                        .filter(instance -> !Strings.isNullOrEmpty(template.getInstanceType()))
+                        .forEach(instance -> instance.getTemplate().setFlavor(template.getInstanceType()));
+            }
+        });
+        return cloudStack;
```

Comment:
like this:
```
    public CloudStack updateWithVerticalScaleRequest(CloudStack cloudStack, FreeIpaVerticalScaleRequest request) {
        InstanceTemplateRequest template = request.getTemplate();
        if (template != null) {
            cloudStack.getGroups().stream()
                    .filter(group -> group.getName().equals(request.getGroup()))
                    .flatMap(group -> group.getInstances().stream())
                    .filter(instance -> !Strings.isNullOrEmpty(template.getInstanceType()))
                    .forEach(instance -> instance.getTemplate().setFlavor(template.getInstanceType()));
        }
        return cloudStack;
    }
```
so you won't have the `request.getTemplate()` call for every group and the null check won't happen in every iteration